### PR TITLE
[Intel MKL] Adding MklTanh op

### PR DESCRIPTION
**Pull request by nhasabni**
This PR adds Tanh op to MKL CPU backend.

### CODE REVIEW

**Comment by gbaned** [comment]
@penpornk Can you please take a look on this PR ? Thanks!

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for your PR and sorry for my delay! I also appreciate the detailed comments in the code.

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc [hunk 1]
```diff
@@ -2949,6 +2949,66 @@ TEST_F(MklLayoutPassTest, NodeRewrite_LeakyReluLeakyReluGrad_Positive) {
...
+      " input: ['A'] }"                                                        \
+      "node { name: 'C' op: 'TanhGrad'"                                        \
+      " attr { key: 'T'                value { type: " #T " } }"               \
+      " input: ['B', 'A'] }"                                                   \
+      "node { name: 'D' op: 'Zeta' attr { key: 'T' value { type: " #T " } }"   \
+      " input: ['A', 'C'] }");                                                 \
+  EXPECT_EQ(DoMklLayoutOptimizationPass(),                                     \
+            "A(" #INPUT ");B(_MklTanh);C(_MklTanhGrad);D(Zeta);DMT/_0(Const);" \
+            "DMT/_1(Const)|A->B;A->C:1;A->D;A:control->DMT/_0:control;"        \
+            "B->C;B:1->C:2;B:control->DMT/_1:control;C->D:1;DMT/_0->B:1;"      \
```
[the remark this answers is not in the document]
Comment (nhasabni) [reply tensorflow/core/common_runtime/mkl_layout_pass_test.cc:3005]: B:1->C:2 carries meta-tensor output for actual tensor output carried by B->C.

**Comment by gbaned** [comment]
@nhasabni Can you please check @penpornk's comments and keep us posted. Thanks!

File: tensorflow/core/kernels/mkl_relu_op_test.cc [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+#include "mkldnn.hpp"
+#include "tensorflow/core/util/mkl_util.h"
+
+// Compare performance of default Tensorflow convolution kernels (Eigen) with
+// MKL kernels on CPU.
+
+// Before running these benchmarks configure OpenMP environment variables:
+//   export KMP_BLOCKTIME=0
+//   export OMP_NUM_THREADS=${num_threads}
```
[the remark this answers is not in the document]
Comment (penpornk) [reply tensorflow/core/kernels/mkl_relu_op_test.cc:44]: Got it. I see that `mkl_conv_ops_test.cc` also has this problem. I'll fix them both later then. (If I can get to it before we fully switch to custom thread pool.)

File: tensorflow/core/common_runtime/mkl_layout_pass_test.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (penpornk) [reply tensorflow/core/common_runtime/mkl_layout_pass_test.cc:3079]: Thank you for the clarification!

**Review by penpornk — APPROVED** [review]
Thank you again for the PR!

**Comment by nhasabni** [comment]
> Thank you again for the PR!

Thanks for review and approval!

**Pull request merged**